### PR TITLE
pump client: support change select pump's strategy

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
support change select pump's strategy
proposal: https://github.com/pingcap/tidb/issues/9201

### What is changed and how it works?
create a new selector when change strategy

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

File: tidb-binlog/pump_client/client.go [hunk 1]
```diff
@@ -241,14 +242,19 @@ func (c *PumpsClient) getPumpStatus(pctx context.Context) (revision int64, err e
 
 // WriteBinlog writes binlog to a situable pump. Tips: will never return error for commit/rollback binlog.
 func (c *PumpsClient) WriteBinlog(binlog *pb.Binlog) error {
+	c.RLock()
+	pumpNum := len(c.Pumps.AvaliablePumps)
+	selector := c.Selector
```
Comment (IANTHEREAL) [inline tidb-binlog/pump_client/client.go:245]: why putting it into a lock scope？

Comment (IANTHEREAL) [reply tidb-binlog/pump_client/client.go:245]: Ok，I think you can put lock in selector, and provide an update function. it's better

**Comment by IANTHEREAL** [comment]
Rest LGTM

**Comment by IANTHEREAL** [comment]
@july2993 Please take a careful look

File: tidb-binlog/pump_client/client.go [hunk 2]
```diff
@@ -416,13 +418,21 @@ func (c *PumpsClient) addPump(pump *PumpStatus, updateSelector bool) {
...
 	}
 
-	c.Pumps.Unlock()
+	c.Unlock()
+}
+
+// SetSelectStrategy sets the selector's strategy, strategy should be 'range' or 'hash' now.
+func (c *PumpsClient) SetSelectStrategy(strategy string) {
+	c.Lock()
+	c.Selector = NewSelector(strategy)
```
Comment (july2993) [inline tidb-binlog/pump_client/client.go:429]: will you check which is really supported before call SetSelectStrategy? like setting `score` must has some problem now.
better return err for wrong strategy and don't fallback the default strategy.

**Review by july2993 — COMMENTED** [review]
rest LGTM
- you should try avoid  global var tsMap, selectorLock which is shard by multi client.
- currently support `range` and `hash`, but i think no different for usage, maybe just confuse people, test more about `hash` if expose this to user.

File: tidb-binlog/pump_client/client.go (hunk 1 above)
Comment (WangXiangUSTC) [reply tidb-binlog/pump_client/client.go:245]: when change strategy will create a new selector. And selector is a member of pump client. so I think lock pump client seems better

**Comment by WangXiangUSTC** [comment]
1. if not use a global tsMap, need set tsMap to new selector from old selector, seems it is a little tedious.
2. when user change the strategy, and at the moment also have write binlog request, will have two selector, and the tsMap should use same one. 
3. one tidb instance should only have one pump client, so it dosen't matter

**Comment by july2993** [comment]
i know it it doesn't matter now, you should encapsulate such thinks in your `PumpsClient`,  but i am ok with this currently.

**Review by july2993 — COMMENTED** [review]
LGTM

**Comment by IANTHEREAL** [comment]
In addition to the poor packaging, LGTM

**Comment by WangXiangUSTC** [comment]
OK, I will refactor these code later. @GregoryIan

**Pull request merged**